### PR TITLE
Fixes #3133

### LINKED ISSUE #3133: ItemsAdder items transferred via cargo
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

Slimefun cargo adds texture to the renamed item

### 📑 Reproduction Steps

I have in the ItemsAdder item named Coin with a texture on the stick. image
I'll put the coin in the first chest
I'll put a stick renamed to Coin in the second chest
I'll have the stick renamed Coin transferred via cargo
The stick acquires the texture of the coin
(Coin (ENG) = Mince (CZ))

We tried it with other items and it works like this with everything.

### 💡 Expected Behavior

The item transferred through the cargo does not acquire texture

### 📷 Screenshots / Videos

The videos were more than 10MB and could not be uploaded, I give a link to youtube
https://youtu.be/I1baDR_b1YQ
https://youtu.be/1-mUxS-7ADA

### 📜 Server Log

N/A

### 📂 `/error-reports/` folder

N/A

### 💻 Server Software

Purpur

### 🎮 Minecraft Version

1.16.x

### ⭐ Slimefun version

image

### 🧭 Other plugins

image

### PR DESCRIPTION
**Pull request by Sfiguz7**
## Description

Check #3133 for details. Cargo would not check for CustomModelData so different ModelData items with the same name and lore would stack when sent via cargo, when they should not.
## Proposed changes

- Added an additional check for CustomModelData: this will help no matter the resource pack, and in particular will fix the issue with ItemsAdder.
## Related Issues (if applicable)

Resolves #3133
## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Review by WalshyDev — APPROVED** [review]
REEE didn't mean to do this

File: src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java [hunk 1]
```diff
@@ -319,17 +321,23 @@ private static boolean equalsItemMeta(@Nonnull ItemMeta itemMeta, @Nonnull ItemM
...
-            return true;
-        } else {
+        } else if (checkLore) {
             Optional<List<String>> itemLore = meta.getLore();
 
-            if (itemMeta.hasLore() && itemLore.isPresent()) {
-                return equalsLore(itemMeta.getLore(), itemLore.get());
-            } else {
-                return !itemMeta.hasLore() && !itemLore.isPresent();
+            if (itemMeta.hasLore() && itemLore.isPresent() && !itemMeta.getLore().equals(itemLore.get())) {
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java:327]: Why is this `.equals` and the old `equalsLore`?

**Review by WalshyDev — CHANGES_REQUESTED** [review]

**Comment by Sfiguz7** [comment]
> REEE didn't mean to do this

TOO LATE HA (I am not sure why it's now equals, let me check what fancy stuff equalsLore does...)

**Review by WalshyDev — DISMISSED** [review]
Alright looks good now

File: src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java [hunk 2]
```diff
@@ -319,17 +321,23 @@ private static boolean equalsItemMeta(@Nonnull ItemMeta itemMeta, @Nonnull ItemM
...
-                return !itemMeta.hasLore() && !itemLore.isPresent();
+            if (itemMeta.hasLore() && itemLore.isPresent() && !equalsLore(itemMeta.getLore(), itemLore.get())) {
+                return false;
+            } else if (itemMeta.hasLore() != itemLore.isPresent()) {
+                return false;
             }
         }
+
+        // Fixes #3133: name and lore are not enough
+        OptionalInt itemCustomModelData = meta.getCustomModelData();
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java:335]: While we are at it, `meta` should be renamed to `snapshot`, now that the class was renamed.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Looks good (as far as I can tell). Just one variable name is a bit confusing (not your fault).

However, we need to make sure that custom model data is NEVER checked if its a Slimefun item. So, are we positive that this is the case?

File: src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java (hunk 2 above)
Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java:335]: > Just one variable name is a bit confusing (not your fault).
Which variable and what would you like the name to be?
I can rename meta easily.

As for CustomModelData, the equalsItemMeta method I have modified is only ever called inside isItemSimilar, and only after a check for the IDs has been performed (lines 281-285 or, due to the cargo filter issue with wrappers, 290-300) - I double checked :)

**Review by svr333 — APPROVED** [review]
lgtm

**Review by WalshyDev — APPROVED** [review]
God these methods are a mess, this LGTM though

**Review by TheBusyBiscuit — APPROVED** [review]

**Pull request merged**